pkg/controller/operconfig/ingressconfig_controller.go

```diff
@@ -0,0 +1,115 @@
+package operconfig
+
+import (
+	"context"
+	"log"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// newIngressConfigReconciler returns a new reconcile.Reconciler
+func newIngressConfigReconciler(client client.Client) *ReconcileIngressConfigs {
+	return &ReconcileIngressConfigs{client: client}
+}
+
+var _ reconcile.Reconciler = &ReconcileIngressConfigs{}
+
+// ReconcileIngressConfigs watches for updates to ingress controller configuration
+// and sets the network policy related labels on the openshift-host-network namespace
+type ReconcileIngressConfigs struct {
+	client client.Client
+}
+
+// constants for namespace and custom resource names
+// namespace in which ingress controller objects are created
+const IngressControllerNamespace = "openshift-ingress-operator"
+
+// namespace representing host network traffic
+// this is also the namespace where to set the ingress label
+const HostNetworkNamespace = "openshift-host-network"
+
+// label to put on the host network namespace
+const HostNetworkLabel = "network.openshift.io/policy-group"
+
+// label value to indicate use of host network ingress policy
+const HostNetworkLabelValue = "ingress"
+
+// default ingress controller name
+const DefaultIngressControllerName = "default"
+
+// Reconcile sets the openshift-host-network namespaces' labels as per the
+// endpointPublishingStrategy of the `default` ingress controller object.
+func (r *ReconcileIngressConfigs) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.Namespace == IngressControllerNamespace && request.Name == DefaultIngressControllerName {
```

Comment:
hey @Miciah can there be more than one IngressController? Could they have different publishing strategies?

Reply:
I think from what @Miciah mentioned the answer is yes. But we decided that for now, we will only do the labeling of the namespace with `policy-group=ingress` for the default controller since that covers 80% of our customer use-caes and for other controllers the user may create (the remaining 20%), we can let them use the `policy-group=host-network` default label available on the namespace as a selector for network policies.